lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -3,8 +3,18 @@
 
   def proxy
     @proxy ||= begin
-      r = request(:get, 'api/v2/smart_proxies', :search => %{name="#{resource[:name]}"})
-      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved: #{error_message(r)}") unless success?(r)
+      path = 'api/v2/smart_proxies'
+      r = request(:get, path, :search => %{name="#{resource[:name]}"})
+
+      unless success?(r)
+        error_string = <<~ERROR.strip
+          Error making GET request to #{request_uri(path)}
+          SmartProxy #{resource[:name]} cannot be retrieved from Foreman:
```

Comment:
I'd try to avoid multiple lines as that's hard to read in the output.

Reply:
Just as you were posting this, I was thinking we need one more line to improve the readability.

The current design is

```
    2021-01-15 16:49:43 [NOTICE] [configure] 1600 out of 1684 done.
    2021-01-15 16:49:48 [ERROR ] [configure] /Stage[main]/Foreman_proxy::Register/Foreman_smartproxy[centos7-foreman-proxy-nightly.mercury.example.com]: Could not evaluate: Error making GET request to https://centos7-katello-3-18.mercury.example.com/api/v2/smart_proxies
    2021-01-15 16:49:48 [ERROR ] [configure] SmartProxy centos7-foreman-proxy-nightly.mercury.example.com cannot be retrieved from Foreman:
    2021-01-15 16:49:48 [ERROR ] [configure] 401 Unauthorized Request: Check credentials for validity
```

I think it should be (taking your other feedback about ordering into consideration) more like:

```

    2021-01-15 16:49:43 [NOTICE] [configure] 1600 out of 1684 done.
    2021-01-15 16:49:48 [ERROR ] [configure] /Stage[main]/Foreman_proxy::Register/Foreman_smartproxy[centos7-foreman-proxy-nightly.mercury.example.com]: Could not evaluate:
    2021-01-15 16:49:48 [ERROR ] [configure] SmartProxy centos7-foreman-proxy-nightly.mercury.example.com cannot be retrieved from Foreman:
    2021-01-15 16:49:48 [ERROR ] [configure] Error making GET request to https://centos7-katello-3-18.mercury.example.com/api/v2/smart_proxies
    2021-01-15 16:49:48 [ERROR ] [configure] 401 Unauthorized Request: Check credentials for validity
```

I think that provides the best clarity, precision, and readability. Let me know if it's agreeable to you